schemas/v3.0/schema.yaml

```diff
@@ -0,0 +1,1400 @@
+        type: object
+        patternProperties:
+          '^[a-zA-Z0-9\.\-_]+$':
+            oneOf:
+              - $ref: '#/definitions/Reference'
+              - $ref: '#/definitions/Callback'
+    patternProperties:
+      '^x-': {}
+    additionalProperties: false
+
+  Schema:
+    type: object
+    properties:
+      title:
+        type: string
+      multipleOf:
+        type: number
+        minimum: 0
+        exclusiveMinimum: true
+      maximum:
+        type: number
+      exclusiveMaximum:
+        type: boolean
+        default: false
+      minimum:
+        type: number
+      exclusiveMinimum:
+        type: boolean
+        default: false
+      maxLength:
+        type: integer
+        minimum: 0
+      minLength:
+        type: integer
+        minimum: 0
+        default: 0
+      pattern:
+        type: string
+        format: regex
+      maxItems:
+        type: integer
+        minimum: 0
+      minItems:
+        type: integer
+        minimum: 0
+        default: 0
+      uniqueItems:
+        type: boolean
+        default: false
+      maxProperties:
+        type: integer
+        minimum: 0
+      minProperties:
+        type: integer
+        minimum: 0
+        default: 0
+      required:
+        type: array
+        items:
+          type: string
+        minItems: 1
+        uniqueItems: true
+      enum:
+        type: array
+        items: {}
+        minItems: 1
+        uniqueItems: true
+      type:
+        type: string
+        enum:
+         - array
+         - boolean
+         - integer
+         - number
+         - object
+         - string
+      not:
+        $ref: '#/definitions/Schema'
+      properties:
+        type: object
+        additionalProperties:
+          $ref: '#/definitions/Schema'
+      additionalProperties:
+        oneOf:
+          - $ref: '#/definitions/Schema'
+          - type: boolean
+        default: true
+      description:
+        type: string
+      format:
+        type: string
+      default: {}
+      $ref:
+        type: string
+        format: uri-ref
+      nullable:
+        type: boolean
+        default: false
+      discriminator:
+        $ref: '#/definitions/Discriminator'
+      readOnly:
+        type: boolean
+        default: false
+      writeOnly:
+        type: boolean
+        default: false
+      example: {}
+      externalDocs:
+        $ref: '#/definitions/ExternalDocumentation'
+      deprecated:
+        type: boolean
+        default: false
+      xml:
+        $ref: '#/definitions/XML'
+    patternProperties:
+      '^x-': {}
+    additionalProperties: false        
+  
+  Discriminator:
+    type: object
+    required:
+      - propertyName
+    properties:
+      propertyName:
+        type: string
+      mapping:
+        type: object
+        additionalProperties:
+          type: string
+
+  XML:
+    type: object
+    properties:
+      name:
+        type: string
+      namespace:
+        type: string
+        format: url
```

Comment:
Should this format be `uri-ref` / `uriref` not `url` ?

Reply:
Hi @webron, as far as I can tell, `url` isn't a format that's in the official specification. It sounds like you're implying here that it's like a `uri`, but required to be absolute?

Reply:
Hi @webron, as far as I can tell, url isn't a format that's in the official specification. It sounds like you're implying here that it's like a uri, but required to be absolute?

Reply:
`uri` is the correct format for an absolute URI.
`uriref` (which became `uri-reference` in the next draft so that is really preferred) is for relative URI references.

It would be really great to avoid `url` as having `uri` and `url` is horribly confusing, particularly as RFC 3986 goes out of its way to explain that there is no hard line between `url` and `urn` (e.g. you cannot make the distinction based on the scheme)

Reply:
@webron @MikeRalphson I'm planning to change this to `uri-reference`, even if we are saying that this schema is a draft-wright-*-00 (a.k.a. "draft-05") schema, and that draft used `uriref`.  I'm doing this because:

1.  Literally no one has ever implemented a validator for draft-wright-*-00.  Ever implementation that I'm aware of skipped from draft-04 (draft-zyp-json-schema-04 + draft-fge-json-schema-validation-00) to draft-06 (draft-wright-*-01).
2.  There is no official meta-schema for draft-05, and there never will be, so there is no programmatic way to inform validators that you are using draft-05
3.  Format is extensible, so using `uri-reference` as a format is entirely valid even in draft-05

Any objections?

Reply:
Ideally we could also publish a draft-06 version where `uri-reference` would be recognized, but `format` validation in general is such a hazy weird optional thing that, really, no one should rely on it unless they are using a specific validator and know it's exact behavior.  e.g. some validators try to use a complex regex to validate the `email` format, but at least one popular validator just checks to see if there's an "@" character in it somewhere and doesn't bother with anything else.